datastructures/search_trees/search_tree.py

```diff
@@ -0,0 +1,68 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar, Callable
+from operator import lt
+
+T = TypeVar('T')
+V = TypeVar('V')
+
+
+class Node(Generic[T, V]):
+    def __init__(self, key: T, value: V) -> None:
+        self.key = key
+        self.value = value
+        self.left = None
+        self.right = None
+        self.parent = None
+
+
+class SearchTree(ABC, Generic[T]):
+    def __init__(self, order_relation: Callable[[T, T], bool] = lt) -> None:
+        self.order_relation = order_relation
+        self.tree_size = 0
+        self.root = None            
+
+
+    def _get_min_by_node(self, current_node) -> tuple[Node, V]:
+        if current_node is not None:
+            while current_node.left:
+                current_node = current_node.left
+        return current_node, current_node.value
```

Comment:
könnte das hier nicht potentiell nen NonetypeError werfen?

Reply:
Yes you're right 👍